pytorch_lightning/trainer/trainer.py

```diff
@@ -481,7 +481,8 @@ def __init__(
         # default .predict() loop
         self.predict_loop = PredictionLoop()
 
-        # .validate() and .test() set this when they load a checkpoint
+        # .fit(), .validate() and .test() set this when they load a checkpoint
+        self.fit_ckpt_path: Optional[str] = None
```

Comment:
We are actually considering having a single variable for this instead.

See: https://github.com/PyTorchLightning/pytorch-lightning/issues/10630

Reply:
Great, thanks @carmocca! I think that makes more sense. 

Any idea when that change will land? In the meantime, I will remove this part of the PR. I think @tchaton already is okay with the request to make `BaseFinetuning.__apply_mapping_to_param_groups` protected instead of private.

Reply:
@speediedan would you be able to handle that issue with this PR?

Reply:
Gladly! To clarify, was the decision to make the attribute protected or not? I see both versions in the  [discussion](#10630) but it sounds like a public attribute was last [mentioned](https://github.com/PyTorchLightning/pytorch-lightning/issues/10630#issuecomment-1010448000):

> Since lately we changed the resume arguments to ckpt_path, I would say `trainer.ckpt_path` is most reasonable.

Reply:
You need to

- Deprecate the previous attributes, in the message suggest using:
- Create a read-only, public `trainer.ckpt_path`